aptos-move/framework/aptos-stdlib/sources/math128.move

```diff
@@ -13,7 +13,11 @@ module aptos_std::math128 {
 
     /// Return the average of two.
     public fun average(a: u128, b: u128): u128 {
-        (a + b) / 2
+        if (a < b) {
```

Comment:
neat

Reply:
Just a note that `ensures result == (a + b) / 2;` is provable as well :)

Reply:
Added a spec statement for average

Reply:
can we do 
```suggestion
       (a & b) + ((a ^ b) >> 1) 
```

Reply:
> can we do

Please not. It forces the prover to use bitvector repr of ints (once this is implemented) which is way slower for arithmetics than this bit fiddling. The gas cost is equivalent.

Reply:
> > can we do
> 
> Please not. It forces the prover to use bitvector repr of ints (once this is implemented) which is way slower for arithmetics. The gas cost is equivalent.